omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/PatientLocationFragmentController.java

```diff
@@ -0,0 +1,134 @@
+import org.openmrs.module.queue.api.search.QueueEntrySearchCriteria;
+import org.openmrs.module.queue.model.QueueEntry;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.annotation.FragmentParam;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.fragment.FragmentConfiguration;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class PatientLocationFragmentController {
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    public void controller(@SpringBean("patientService") PatientService patientService,
+                           @SpringBean("adtService") AdtService adtService,
+                           @FragmentParam("app") AppDescriptor app,
+                           UiSessionContext uiSessionContext,
+                           UiUtils ui,
+                           FragmentConfiguration config,
+                           FragmentModel model) throws Exception {
+
+        Object patientConfig = config.get("patient");
+        if (patientConfig == null ) {
+            patientConfig = config.get("patientId");
+        }
+        Patient patient = null;
+        if (patientConfig != null) {
+            if (patientConfig instanceof Patient) {
+                patient = (Patient) patientConfig;
+            }
+            else if (patientConfig instanceof PatientDomainWrapper) {
+                patient = ((PatientDomainWrapper) patientConfig).getPatient();
+            }
+            else if (patientConfig instanceof Integer) {
+                patient = patientService.getPatient((Integer)patientConfig);
+            }
+            else if (patientConfig instanceof String) {
+                patient = patientService.getPatientByUuid((String)patientConfig);
+            }
+        }
+        if (patient == null) {
+            throw new IllegalArgumentException("No patient found. Please pass patient into the configuration");
+        }
+        VisitDomainWrapper activeVisit = adtService.getActiveVisit(patient, uiSessionContext.getSessionLocation());
+        String patientStatus = "";
+        String queueName = "";
+        Location currentInpatientLocation = null;
+        if (activeVisit != null ) {
+            // #1 Check if patient was admitted
+            InpatientAdmissionSearchCriteria criteria = new InpatientAdmissionSearchCriteria();
+            criteria.setPatientIds(Arrays.asList(patient.getPatientId()));
+            criteria.setIncludeDischarged(false);
+            List<InpatientAdmission> inpatientAdmissions = adtService.getInpatientAdmissions(criteria);
+            InpatientRequest currentInpatientRequest = null;
+            if (inpatientAdmissions != null && !inpatientAdmissions.isEmpty()) {
+                for (InpatientAdmission inpatientAdmission : inpatientAdmissions) {
```

Comment:
I think the assumption here is that there should only ever be one inpatient admission per patient, but, of course the query returns a list since it's a search?  If I'm understanding it, I think it would be better just to use inpatientAdmissions.get(0), but log a warning if inpatientAdmissins.size() > 2?

Or you could keep it this way but just add a comment saying there should never be more than one?